test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Formatter/Serialization/ODataResourceSerializerTests.cs

```diff
@@ -892,6 +902,280 @@ public void CreateResource_Works_ToAppendDynamicProperties_ForOpenEntityType()
             Assert.Equal(new DateTimeOffset(dateTime), dateTimeProperty.Value);
         }
 
+        [Fact]
+        public void CreateResource_Works_WithInstanceAnnotations()
+        {
+            // Arrange
+            IEdmModel model = SerializationTestsHelpers.SimpleOpenTypeModel();
+
+            IEdmEntitySet customers = model.EntityContainer.FindEntitySet("Customers");
+
+            IEdmEntityType customerType = model.FindDeclaredType("Default.Customer") as IEdmEntityType;
+            Type simpleOpenCustomer = typeof(SimpleOpenCustomer);
+            model.SetAnnotationValue(customerType, new ClrTypeAnnotation(simpleOpenCustomer));
+
+            IEdmComplexType addressType = model.FindDeclaredType("Default.Address") as IEdmComplexType;
+            Type simpleOpenAddress = typeof(SimpleOpenAddress);
+            model.SetAnnotationValue(addressType, new ClrTypeAnnotation(simpleOpenAddress));
+
+            IEdmEnumType enumType = model.FindDeclaredType("Default.SimpleEnum") as IEdmEnumType;
+            Type simpleEnumType = typeof(SimpleEnum);
+            model.SetAnnotationValue(enumType, new ClrTypeAnnotation(simpleEnumType));
+
+            model.SetAnnotationValue(customerType, new InstanceAnnotationDictionaryAnnotation(
+            simpleOpenCustomer.GetProperty("InstanceAnnotations")));
+
+            var in1 = new Dictionary<string, object>();
+            in1.Add("NS.test1", 123);
+
+            var in2 = new Dictionary<string, object>();
+            in2.Add("NS.test2", 345);
+
+            var instAnn = new Dictionary<string, IDictionary<string, object>>();
+            instAnn.Add("", in1);
+            instAnn.Add("Name", in2);
```

Comment:
>instAnn [](start = 12, length = 7)

test to add an instance annotation using the "Address" value

Reply:
yes; please add positive test for setting an instance annotation on a complex (and a property of a complex) object.

---
In reply to: [450395345](https://github.com/OData/WebApi/pull/2219#discussion_r450395345) [](ancestors = 450395345)